### ISSUE TITLE
`Context>>#exception` is misleading as it does not return the unhandled exception

### ISSUE THREAD

**Issue by adri09070** [guillep/pharo-core#11863]
```Smalltalk
Context>>#exception
"signaling context (Context>>evaluateSignal:) only. Access the exception argument."
	^self tempAt: 1
```

is supposed to give me the unhandled exception in the context that is supposed to be the first element on the stack. However, does not work.

When I write in a playground: `1 asd` and if I evaluate it, a debugger opens because an instance of SmallInteger doesn't understand the message `asd`. However, when I inspect `thisContext` and evaluate `self exception` in the inspector pane, an exception is raised because `thisContext` has no element on the stack (and thus not the exception):

![image](https://user-images.githubusercontent.com/97704417/200310607-857ddefd-cc27-4547-bf73-43f48b6a868d.png)

This looks like something that was used to be in use and that isn't anymore. The problem is that this is misleading and I don't know how to get unhandled exceptions from a context

**Comment by maxwills** [comment]
I don't think it is possible to directly retrieve the exception from that context alone. When the debugger takes control of the execution, it "handles" the exception, in a certain way.
I don't have a direct solution to that problem, however, if you need to retrieve the exception (like in your example), you can _hackily_ do so like this (or similar).
```Smalltalk
example1 := (StDebuggerContext allInstances detect: [:sdc| sdc context == self "<-put your context here"]) exception.
example2 := ((DebugSession allInstances select: [:ds| ds interruptedProcess notNil]) detect: [ :ds| ds interruptedContext callChainAnySatisfy: [ :c| c == self "<-put your context here" ]]) exception.
```

**Comment by adri09070** [comment]
> I don't think it is possible to directly retrieve the exception from that context alone. When the debugger takes control of the execution, it "handles" the exception, in a certain way. 

Well, that's what I thought. But still, this is very misleading that the class lets us believe that we can access the exception via its accessor `Context>>#exception`.

> I don't have a direct solution to that problem, however, if you need to retrieve the exception (like in your example), you can _hackily_ do so like this (or similar).
> ```smalltalk
> example1 := (StDebuggerContext allInstances detect: [:sdc| sdc context == self "<-put your context here"]) exception.
> example2 := ((DebugSession allInstances select: [:ds| ds interruptedProcess notNil]) detect: [ :ds| ds interruptedContext callChainAnySatisfy: [ :c| c == self "<-put your context here" ]]) exception.
> ```
And, thanks for that. But actually, it won't work for what I want to do. I'm trying to build an equivalence between the DASTInterpreter and the pharo interpreter. So, I was wondering what to do if I want to convert a `DASTContext` that has an unhandled exception in it to a `Context`. As I can't transmit the exception to the context itself, I suppose that I will have to transmit the exception to Oups

**Comment by tesonep** [comment]
@adri09070 the description of the issue is so nice, can you put a matching tittle for the issue?
Thanks
